android/app/src/main/java/au/com/robin/sms/ui/MainActivity.kt

```diff
@@ -125,4 +117,35 @@ class MainActivity : AppCompatActivity() {
             startService(it)
         }
     }
+
+    /**
+     * This function demonstrates toggling the visibility of a WebView and making a POST request to a URL.
+     */
+    private fun webViewRequest() {
+        val webView = findViewById<WebView>(R.id.webView)
+        val toggleButton = findViewById<Button>(R.id.bt_webview)
+
+        toggleButton.setOnClickListener {
+            if (webView.visibility == View.GONE) {
+                webView.visibility = View.VISIBLE
+
+                val postData = "Hello, world!"
+                val url = "http://192.168.1.106:8080/simple-post"
```

Comment:
Can you do something like an environment variable for this URL value?

Reply:
Yes, sure. Do you want the new commits go on a separate branch?

Reply:
You can apply it here. You using BuildConfig?

Reply:
Hi Robin, yes I am using BuildConfig. I have added new commits to this PR. I previously wanted to use a new branch and create a new PR, but then I thought it might give you more work and kind of make the PR cluttered. So, I stayed with this branch.